src/utils/token_to_fiat_oracle.ts

```diff
@@ -0,0 +1,133 @@
+import type { FiatOracle } from './fiat_oracle';
+import type { CoinGeckoPriceRequestParams, CoinGeckoPriceResponseData } from './coingecko_types';
+import { TokenFiatPair } from './token_fiat_price';
+import type { FiatID, TokenID } from './fiat_oracle_types';
+
+const pollingIntervalMilliseconds = 1000 * 60 * 15; // 15 min
+
+function cachePairFilterFactory(fiat: FiatID, token: TokenID) {
+	return (p: TokenFiatPair) => p.fiat === fiat && p.token === token;
+}
+
+/*
+ * The CoinGecko wrapper does not support browser.
+ * Here it is the reported feature request for supporting
+ * the browser platform by just removing node dependencies.
+ *
+ * https://github.com/miscavage/CoinGecko-API/issues/19
+ *
+ * When the report is taken and fixed, please:
+ * 1. Import the CoinGecko wrapper.
+ * 2. Define a CoinGecko client as private readonly at the
+ *   constructor arguments. So we can test it.
+ * 3. On each call to the API, replace the fetch for the actual
+ *   API call.
+ *
+ * Examples
+ *
+ * 1. import CoinGecko from 'coingecko-api'
+ * 2. constructor(private readonly client = new CoinGecko()) {}
+ * 3. const response = await this.client.ping() // for the ping() function
+ *    const response = await this.client.simple.price(params) // for getting the price of a currency
+ */
+
+export class TokenToFiatOracle implements FiatOracle {
+	private cachedPrices: TokenFiatPair[] = [];
+	private cacheTimestamp = 0;
+	private syncPromise?: Promise<void>;
+
+	/**
+	 * @param {FiatID[]} fiats The currency IDs to quote arweave in
+	 * @param {TokenID[]} tokens The token ID of the requested coin
+	 * @param {number} cacheLifespan Milliseconds that has to pass until the cache is invalid
+	 * @param {@link fetch} _fetch The standard fetch method. Defined here for testing propuses
+	 * @return {TokenToFiatOracle} A class representing the oracle for the selected currencies
+	 */
+	constructor(
+		private readonly fiats: FiatID[] = ['usd'],
+		private readonly tokens: TokenID[] = ['arweave'],
+		private readonly cacheLifespan = pollingIntervalMilliseconds,
+		private readonly _fetch = fetch
+	) {}
+
+	private get currentlyFetchingPrice(): boolean {
+		return !!this.syncPromise;
+	}
+
+	private get tokensCSV(): string {
+		return this.tokens.join(',');
+	}
+
+	private get fiatCSV(): string {
+		return this.fiats.join(',');
+	}
+
+	/**
+	 * @param {FiatID} fiat The fiat currency ID
+	 * @param {TokenID} token The token ID
+	 * @throws {@link Error} If no such pair cached
+	 * @returns {Promise<TokenToFiatPrice>} The cached price value
+	 */
+	public async getPriceForFiatTokenPair(fiat: FiatID = 'usd', token: TokenID = 'arweave'): Promise<TokenFiatPair> {
```

Comment:
Defaulting to `usd` and `arweave` as the input values in many of these functions will probably hurt readability of the code from a client perspective and also increase the probability of getting the wrong values down the line.

Reply:
I've defined a TokenFiatPair that's only a representation of the pair. Then, `TokenFiatRate` extends it, and adds a `rate` attribute.
At 3e7db20c30611c9f043f3dc981476e8f1ae2d166